security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -46,9 +46,14 @@ var (
 	// with extra SAN (labels, etc) in data path.
 	ProvCert = env.RegisterStringVar("PROV_CERT", "",
 		"Set to a directory containing provisioned certs, for VMs").Get()
+
+	// OutputKeyCertToDir path is set, it will restore the cert from the signed by the CA
+	OutputKeyCertToDir = env.RegisterStringVar("OUTPUT_CERTS", "",
+		"The output directory for the key and certificate. If empty, key and certificate will not be saved. "+
+			"Must be set for VMs using provisioning certificates.").Get()
 )
 
-type citadelClient struct {
+type CitadelClient struct {
```

Comment:
This should not be exported otherwise what's the point of having the Client interface.

And you later did type check in security/pkg/nodeagent/cache/secretcache.go to get the concrete type of the CaClient which is also bad because now you need include all the different type of the ca clients (CitadelClient, GoogleClient and VauleClient) in that file, which again making the interface useless.

Reply:
Yes because in the last version I need to use some specific variable of the concrete type. Right now will refactor it to make it not exported